File: frontend/src/app/domain/models/transaction-history.ts
export type TransactionSnapshot = Record<string, string | null>;

export interface TransactionHistoryEntry {
  id: string;
  transactionId: string;
  operation: 'create' | 'update' | 'delete';
  source: string;
  batchId?: string;
  before?: TransactionSnapshot;
  after?: TransactionSnapshot;
  createdAt: string;
}

File: backend/app/schemas/transaction_history.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TransactionHistoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    transaction_id: UUID
    operation: str
    source: str
    batch_id: UUID | None
    before: dict[str, Any] | None
    after: dict[str, Any] | None
    created_at: datetime


class ImportBatchRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    created_at: datetime
    created_count: int
    remaining_count: int
